daemon/daemon_unix.go

```diff
@@ -402,7 +402,7 @@ func verifyContainerResources(resources *containertypes.Resources, sysInfo *sysi
 		logrus.Warnf("Your kernel does not support CPU cfs period. Period discarded.")
 		resources.CPUPeriod = 0
 	}
-	if resources.CPUPeriod > 0 && (resources.CPUPeriod < 1000 || resources.CPUQuota > 1000000) {
+	if resources.CPUPeriod < 0 || (resources.CPUPeriod > 0 && (resources.CPUPeriod < 1000 || resources.CPUPeriod > 1000000)) {
```

Comment:
I think you can simplify this to:

    if resources.CPUPeriod != 0 && (resources.CPUPeriod < 1000 || resources.CPUPeriod > 1000000) {